contracts/contracts/vault/VaultCore.sol

```diff
@@ -385,32 +385,102 @@ contract VaultCore is VaultInitializer {
      * @return totalUnits Total balance of Vault in units
      */
     function _rebase() internal whenNotRebasePaused returns (uint256) {
-        uint256 ousdSupply = oUSD.totalSupply();
+        uint256 supply = oUSD.totalSupply();
         uint256 vaultValue = _totalValue();
-        if (ousdSupply == 0) {
+        uint256 nonRebasing = oUSD.nonRebasingSupply();
```

Comment:
Fetching `nonRebasing` can be moved to into `_nextYield` function as it is only required in there

Reply:
Awesome! `f1f5d07eb86c4af92ed30d8aa15874b77456d6d8`